lib/matplotlib/axes/_base.py

```diff
@@ -2871,8 +2871,12 @@ def set_xlim(self, left=None, right=None, emit=True, auto=False, **kw):
 
         self._process_unit_info(xdata=(left, right))
         if left is not None:
+            if isinstance(left,float) and (math.isnan(left) or np.isinf(left)):
```

Comment:
think of it this way. We recently discovered some bugs that were because numpy ints and python ints aren't the same in py3k, and so `isinstance(a, int)` of a numpy int wouldn't work anymore. Better safe than sorry.